### PR TITLE
Also retry all-nodes requests on I/O error

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
a005c1d3d2f3bb248af229502c5d8cebfa9ab219 inadvertently altered the
reconnection behaviour of all-nodes requests by only reconnecting after an
error event in epoll. We (obviously) also need to reconnect after a normal
connection I/O error (e.g. broken pipe).

### CODE REVIEW

**Review by david-eckardt-sociomantic — APPROVED** [review]

File: src/swarm/neo/client/mixins/AllNodesRequestCore.d [hunk 1]
```diff
@@ -140,8 +141,14 @@ public struct AllNodesRequest ( Request, Connector, Disconnected, Initialiser,
 
                 this.handler();
             }
-            // Only retry in the case of a connection error. Other errors
-            // indicate internal problems and should not be retried.
+            // Only retry in the case of a connection I/O error or error event
+            // in epoll. Other errors indicate internal problems and should not
+            // be retried.
+            catch ( IOError e )
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: Huh, I thought IOError was inherited from EpollException, at least that was my understanding at the time.

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: Here, so `IOError` inherits from `IOWarning` and `IOWarning` from `EpollException`: https://github.com/sociomantic-tsunami/ocean/blob/ba54399a79812c2be4c135a0597b88a93113f568/src/ocean/io/select/protocol/generic/ErrnoIOException.d#L75 https://github.com/sociomantic-tsunami/ocean/blob/ba54399a79812c2be4c135a0597b88a93113f568/src/ocean/io/select/protocol/generic/ErrnoIOException.d#L39

Comment (david-eckardt-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: `IOWarning` is a subclass of `ErnoException`.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: `ErrnoException`

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: :man_facepalming:

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: Still, there's a question about `IOWarning`. `IOWarning` should indicate remote party hangup. I think in this case we should reconnect as well.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/mixins/AllNodesRequestCore.d:147]: Hmm, the [description](https://github.com/sociomantic-tsunami/ocean/blob/ba54399a79812c2be4c135a0597b88a93113f568/src/ocean/io/select/protocol/generic/ErrnoIOException.d#L34-L35) of `IOWarning` doesn't sound like that, but I'm not sure where it's actually thrown, in practice.

**Pull request merged**